### ISSUE TITLE
BasicDao isn't usable with Spring Annotations [moved]

### ISSUE THREAD

**Issue by jyemin** [MorphiaOrg/morphia#187]
This is [Issue 187](http://code.google.com/p/morphia/issues/detail?id=187) moved from a Google Code project.
Added by 2010-12-14T23:10:15.000Z by [serp...@gmail.com](http://code.google.com/u/100579738380910599171/).
Please review that bug for more context and additional comments, but update this bug.
 Closed (WontFix).

Original labels: Type-Defect, Priority-Low
### Original description

```
At the moment to build BasicDao I need to provide Mongo and Morphia objects. This becomes a problem in an app that uses Spring Annotations to eliminate the need to declare DAOs in app context.

Propose to remove any real initialization of BasicDao from constructor. 

Instead, I should be able to set all the dependencies via setter methods and only after that call some method that will initialize the guts of BasicDao.
```
